File: PerftestsJSAndroid/assets/www/javascripts/stopwatch.js
function StopWatch() {
	this.start();
}
StopWatch.prototype.start = function(o) {
	this.t0 = new Date().getTime();
};
StopWatch.prototype.stop = function(o) {
	var t1 = new Date().getTime();
	return "time: \t"+(t1-this.t0)+" ms";
};

File: PerftestsAndroid/src/de/objectcode/perftests/utils/StopWatch.java
package de.objectcode.perftests.utils;

public class StopWatch {
	
	private Measurement m0;
	private Measurement m1;
	
	class Measurement {
		long tNs;
		long tMs;
		long heapTotal;
		public Measurement() {
			tNs = System.nanoTime();
			tMs = System.currentTimeMillis();
			heapTotal = Runtime.getRuntime().totalMemory();
		}
		public String diff(Measurement o) {
			StringBuilder b = new StringBuilder();
			b.append("time: \t").append((tNs-o.tNs)/1000000).append(" ms\n");
			b.append("heapTotal: \t").append(heapTotal-o.heapTotal).append(" bytes\n");
			return b.toString();
		}
	}
	
	public StopWatch() {
		start();
	}
	
	public void start() {
		m0 = new Measurement();
	}

	public String stop() {
		m1 = new Measurement();
		return m1.diff(m0);
	}
}
